Source/Dafny/DafnyAst.cs

```diff
@@ -10607,6 +10621,30 @@ public class CasePattern<VT> where VT : IVariable
       Id = bv.Name;
       Var = bv;
     }
+  public override string ToString(){
+    Contract.Requires(this != null);
+    var v = this.Var;
+    string c = "";
+    if (v != null) {
+      c+=(v.DisplayName);
+      if (v.OptionalType is NonProxyType || DafnyOptions.O.DafnyPrintResolvedFile != null) {
+        c+=(": "+v.OptionalType.TypeName(null, true));
+      }
+      } else {
+        if (this.Id.StartsWith(BuiltIns.TupleTypeCtorNamePrefix)) {
+          Contract.Assert(this.Arguments != null);
+        } else {
+          c+=(this.Id);
+        }
+        if (this.Arguments != null) {
+          c+=("(");
+          List<String> argStr = this.Arguments.ConvertAll<String>(arg => arg.ToString());
+          c+= String.Join(",", argStr);
+          c+=")";
```

Comment:
Put spaces around operators `+=` and around `+`.
No need for parentheses on the right-hand sides of lines 10629, 10631, 10637, 10640, 10643.
If you wish, you can replace `List<String>` on L10641 with `var`.